cpp/include/kvikio/utils.hpp

```diff
@@ -28,6 +28,20 @@
 #include <kvikio/error.hpp>
 #include <kvikio/shim/cuda.hpp>
 
+// Macros used for defining symbol visibility, only GLIBC is supported.
+// Since KvikIO is header-only, we rely on the linker to disambiguate inline functions
+// that have (or return) static references. To do this, the relevant function must have
+// `__attribute__((visibility("default")))`. If not, then if KvikIO is used in two
+// different DSOs, the function will appear twice, and there will be two static objects.
+// See <https://github.com/rapidsai/kvikio/issues/442>.
+#if (defined(__GNUC__) && !defined(__MINGW32__) && !defined(__MINGW64__))
```

Comment:
*nit*: Can we also support clang (https://clang.llvm.org/docs/AttributeReference.html#visibility) given the various people compiling on macos?

Reply:
Actually, clang also defines `__GNUC__` by default but I guess adding `defined(__clang__)` is good since `__attribute__((visibility(...)))` is valid clang code as well.